glow/rate_limiter_test.go

```diff
@@ -0,0 +1,84 @@
+package glow
+
+import (
+	"testing"
+	"time"
+)
+
+func TestRateLimiterSingle(t *testing.T) {
+	rl := NewRateLimiter(1, 50*time.Millisecond)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+
+	count := 0
+	allowed := 0
+	for _ = range ticker.C {
+		if rl.Allow() {
+			allowed++
+		}
+		count++
+		if count == 4 {
+			break
+		}
+	}
+
+	t.Logf("allowed %v", allowed)
+
+	if allowed != 2 {
+		t.Errorf("Single test failed, wanted %v got %v", 2, allowed)
```

Comment:
this check totally eliminates any need to log the value; if it's anything but 2 you'll get an error, and the error itself will tell you what non-2 value happened

Reply:
Agree